platform/engineering/design-docs/2020-08-17-mobile-api-design-doc.md

```diff
@@ -0,0 +1,173 @@
+The mobile `ApplicationController` would extend the core `ApplicationController` and override its authentication methods.
+
+Enabling the mobile API will expose some tech debt in the `ApplicationController` that is not a requirement to fix; but it would clean up things for all its sub-classes.
+The logic in the core `ApplicationController` that does not apply globally could move to concerns be and 
+included as needed. e.g. `render_job_id` only applies to the `DocumentsController` and the `EVSSClaimsController`. 
+
+#### Routing
+
+The mobile engine's endpoints mount at `/mobile`:
+
+```ruby
+mount Mobile::Engine, at: '/mobile'
+```
+
+Some endpoints such as facilities may not require optimization. They will pass-through via routes:
+
+```ruby
+# modules/mobile/config/routes.rb (redirect starts at the app root so this is /v0/mobile/.. pointing to )
+get '/v0/facilities/va/:id' => redirect('/v1/facilities/va/%{id}')
+```
+
+#### Request/response differences
+
+How requests and responses function within the mobile module will be isolated from the core API. However, there are improvements that could
+be ported back and it would be good to get consensus on an approach. The plan is for views, such as user profile, that currently over-fetch 
+or over-deliver, should implement fieldsets and includes in REST url params so no extra data is returned. JSON-API defines
+sparse [fieldsets](https://jsonapi.org/format/#fetching-sparse-fieldsets) and [includes](https://jsonapi.org/format/#fetching-includes) which
+can be implemented with custom code or through gems such as [JSON-API Resources](https://jsonapi-resources.com/) or [Graphiti](https://www.graphiti.dev/guides/).
+
+
+### Code Location
+The code will live in a 'mobile' directory within the [modules](https://github.com/department-of-veterans-affairs/vets-api/tree/master/modules) directory within vets-api.
+
+### Testing Plan
+To ensure changes to shared backend services will not break code, tests will be written to assert the mobile API requirements. Any breaking changes to upstream services will fail our tests.
+
+### Logging
+We plan to adhere to all VSP standards for logging.
+
+### Debugging
+TBD
+
+### Caveats
+TBD
+
+### Security Concerns
+#### Potential threats
+
+##### SSOe introspect DoS
+Adding more surface area to the API does enable new possibilities for denial-of-service attacks. We plan on making a call to SSOe (introspect) with every vets-api session cache miss, to validate the access token and establish a session. An attacker could DoS the introspect endpoint by passing multiple requests all with invalid access tokens.
+
+Remediation could include caching each “miss” for a period of time, protecting the introspect endpoint.
+
+##### CSRF for API POST requests
+We know we can disable CSRF checking on the server-side for the mobile POST routes, but that seems unwide.
```

Comment:
it should be OK to only enforce CSRF prevention when cookie-based authentication is used

if you're authenticating with a mobile auth token (presumably passed via HTTP Header? I haven't read the other design doc yet), then CSRF prevention isn't necessary